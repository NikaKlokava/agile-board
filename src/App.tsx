import { Route, Routes } from "react-router";
import "./index.css";
import { BoardPage } from "./pages/board";
import { ErrorPage } from "./pages/error";

function App() {
  return (
    <Routes>
      <Route path="/agile-board" element={<BoardPage />} />
      <Route path="*" element={<ErrorPage />} />
    </Routes>
  );
}

export default App;
