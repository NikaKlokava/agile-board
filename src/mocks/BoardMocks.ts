type MockSubtaskType = {
  name: string;
};

type MockTaskType = {
  name: string;
  description: string;
  subtasks: Array<MockSubtaskType>;
};

export const MockTask: MockTaskType = {
  name: "Create something",
  description: "... it's very important ...",
  subtasks: [
    { name: "while doing 1..." },
    { name: "while doing 2..." },
    { name: "while doing 3..." },
  ],
};

export const MockBoards = [
  {
    name: "Platform Launch",
    board_column: [
      { title: "Todo", tasks: Array.from({ length: 4 }, () => MockTask) },
      { title: "Doing", tasks: Array.from({ length: 6 }, () => MockTask) },
      { title: "Done", tasks: Array.from({ length: 2 }, () => MockTask) },
    ],
  },
  {
    name: "Marketing Plan",
    board_column: [
      { title: "Todo", tasks: Array.from({ length: 7 }, () => MockTask) },
      { title: "Doing", tasks: Array.from({ length: 2 }, () => MockTask) },
      { title: "Done", tasks: Array.from({ length: 4 }, () => MockTask) },
    ],
  },
  {
    name: "Roadmap",
    board_column: [
      { title: "Now", tasks: Array.from({ length: 4 }, () => MockTask) },
      { title: "Next", tasks: Array.from({ length: 6 }, () => MockTask) },
    ],
  },
];
