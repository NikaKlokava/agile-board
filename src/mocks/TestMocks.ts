export const MockTestAddBoard = {
  name: "test name",
  columns: [{ title: "test title", uuid: "test uuid" }],
  usersEmail: ["test"],
  time: 0,
  uuid: "test",
};
