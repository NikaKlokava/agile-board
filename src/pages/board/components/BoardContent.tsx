import { EditBoardModal, TaskModal } from "./modals";
import { useState } from "react";
import { Sidebar } from "./Sidebar";
import cl from "./styles/board_content.module.css";

export const BoardContent = () => {
  const [editBoardVisible, setEditBoardVisible] = useState<boolean>(false);
  const [taskModalVisile, setTaskModalVisile] = useState<boolean>(false);

  return (
    <>
      <Sidebar />
      <div className={cl.board_content_wrapper}>
        {/* {data[0].board_column.map((column, index) => {
          return (
            <div className={cl.content_column} key={index}>
              <div className={cl.title_container}>
                <div className={cl.column_circle} />
                <div className={cl.column_title}>{column.title}</div>
              </div>
              {column.tasks.map((task, index) => {
                return (
                  <div
                    className={cl.tasks_container}
                    key={index}
                    data-testid="task-container"
                    onClick={() => setTaskModalVisile(true)}
                  >
                    <div className={cl.task_title}>{task.name}</div>
                    <div
                      className={cl.task_success}
                    >{`0 of 5 completed tasks`}</div>
                  </div>
                );
              })}
            </div>
          );
        })} */}
        <div
          className={cl.add_column}
          onClick={() => setEditBoardVisible(true)}
          data-testid="add_column"
        >
          <p className={cl.add_column_title}>{"+ New Column"}</p>
        </div>
        {editBoardVisible && (
          <EditBoardModal onWrapperClick={() => setEditBoardVisible(false)} />
        )}
        {taskModalVisile && (
          <TaskModal onWrapperClick={() => setTaskModalVisile(false)} />
        )}
      </div>
    </>
  );
};
