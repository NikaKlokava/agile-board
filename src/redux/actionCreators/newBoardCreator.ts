export const addBoardName = ({ name }: BoardNameType) => {
  return {
    type: "board/addBoardName",
    payload: {
      name,
    },
  };
};

export const addColumns = ({ board_columns }: ColumnsType) => {
  return {
    type: "board/addColumns",
    payload: {
      board_columns: board_columns,
    },
  };
};
