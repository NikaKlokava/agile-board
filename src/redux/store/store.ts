import { legacy_createStore } from "redux";
import { boardReducer } from "../reducers/boardReducer";

const store = legacy_createStore(boardReducer);

export default store;
