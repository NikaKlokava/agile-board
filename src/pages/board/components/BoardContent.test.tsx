import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { act } from "react-dom/test-utils";
import renderer from "react-test-renderer";
import App from "../../../App";
import { BoardContent } from "./BoardContent";
import { Provider } from "react-redux";
import store from "../../../redux/store/store";
import { addBoard } from "../../../redux/actionCreators/newBoardCreator";
import { MockTestAddBoard } from "../../../mocks/TestMocks";

describe("Test the BoardContent component", () => {
  test("The BoardContent renders correctly", () => {
    const boardContentSnap = renderer
      .create(
        <Provider store={store}>
          <BoardContent />
        </Provider>
      )
      .toJSON();
    expect(boardContentSnap).toMatchSnapshot();
  });
});

describe("Test the New Column element", () => {
  test("The New Column element should not be visible if we have no active board", () => {
    render(
      <MemoryRouter initialEntries={["/agile-board"]}>
        <App />
      </MemoryRouter>
    );
    const newColumnEl = screen.queryByTestId("add_column");

    expect(newColumnEl).not.toBeInTheDocument();
  });

  test("The New Column element should be visible if we have active board", () => {
    render(
      <MemoryRouter initialEntries={["/agile-board"]}>
        <App />
      </MemoryRouter>
    );

    act(() => {
      store.dispatch(addBoard(MockTestAddBoard));
    });

    const newColumnEl = screen.queryByTestId("add_column");

    expect(newColumnEl).toBeInTheDocument();
  });
});

describe("Test the Task Container", () => {
  test("The task container should not be visible if there is no task", () => {
    render(
      <MemoryRouter initialEntries={["/agile-board"]}>
        <App />
      </MemoryRouter>
    );
    const taskContainerElements = screen.queryAllByTestId("task-container");

    taskContainerElements.forEach((taskElement) => {
      expect(taskElement).not.toBeInTheDocument();
    });
  });
});
