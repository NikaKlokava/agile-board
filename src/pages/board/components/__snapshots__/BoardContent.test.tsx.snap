// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the BoardContent component The BoardContent renders correctly 1`] = `
Array [
  <div
    className="sidebar_wrapper"
  >
    <div
      className="sidebar_title"
    >
      ALL BOARDS (0)
    </div>
    <div
      className="board_title_items"
    >
      <div
        className="title_item"
        data-testid="new-board"
        onClick={[Function]}
      >
        <div
          className="icon_new_board"
        />
        <p
          className="title new_board"
        >
          New board
        </p>
      </div>
    </div>
    <div
      className="hide_sidebar"
    >
      <div
        className="hide_icon"
      />
      <div
        className="hide_text"
      >
        Hide sidebar
      </div>
    </div>
  </div>,
  <div
    className="board_content_wrapper"
  >
    <div
      className="add_column"
      data-testid="add_column"
      onClick={[Function]}
    >
      <p
        className="add_column_title"
      >
        + New Column
      </p>
    </div>
  </div>,
]
`;
